### ISSUE TITLE
OS X Build error , not linking boost libraries

### ISSUE THREAD

**Issue by mike2lab** [valiant1x/intensecoinwallet#41]
When i try to build the source files I get this error:

boost/thread/v2/thread.hpp' file not found

How to fix this ?

**Comment by mike2lab** [comment]
yes i've installed boost through the brew but still get that error.
I just copied the boost folder from /usr/local/Cellar/boost/1.67.0_1/include/boost/ and pasted into the /intensecoinwallet/intense/contrib/epee/include/ folder but still the same error:

**Comment by iedemam** [comment]
You shouldn't move boost into that location. Delete what you've added and/or start with a fresh clone of the source. Then run ./build.sh and post the output here.

Which version of macOS are you using?

**Comment by mike2lab** [comment]
Looks like there's not that file in that folder boost/thread/v2/thread.hpp.
I can't see it

**Comment by mike2lab** [comment]
The output is that only error >>
 fatal error: 'boost/thread/v2/thread.hpp' file not found >> #include <boost/thread/v2/thread.hpp>

**Comment by iedemam** [comment]
Looks like Boost 1.67 moves the location of some headers and our build system doesn't know about that yet. I found an upstream mention of it in here: https://github.com/monero-project/monero/issues/3663

For now, could you install an older version of boost instead of 1.67? Something like:

```
$ brew uninstall boost
Uninstalling /usr/local/Cellar/boost/1.66.0... (13,101 files, 435.3MB)

$ brew search boost
==> Searching local taps...
boost                       boost-build                 boost-python                boost-python@1.59           boost@1.57                  boost@1.60
boost-bcp                   boost-mpi                   boost-python3               boost@1.55                  boost@1.59
==> Searching taps on GitHub...
caskroom/cask/turbo-boost-switcher
==> Searching blacklisted, migrated and deleted formulae...

$ brew install boost@1.60

**Comment by mike2lab** [comment]
Ok i did that and now i get  the error:

CMake Error at CMakeLists.txt:48 (message):
  Could not find Boost libraries, please make sure you have installed
  Boost or libboost-all-dev (1.58) or the equivalent

**Comment by iedemam** [comment]
Did you make clean before rebuilding?
Or better yet, make a fresh clone of the repo and build fresh from there to make sure nothing else was copied in or left behind.

**Comment by mike2lab** [comment]
Yes I did it from scratch  again and also checked that i've installed boost 1.60 but i get that error:
Could not find Boost libraries, please make sure you have installed
  Boost or libboost-all-dev (1.58) or the equivalent

**Comment by iedemam** [comment]
OK, thanks for the report. I can replicate that here too: 1.60 isn't recognized and 1.67 fails to build. I'll work on fixing the 1.67 incompatibility.

BTW, which version of macOS are you using?

**Comment by mike2lab** [comment]
I just reinstalled boost 1.67 and works like before until i get the error : 
fatal error: 'boost/thread/v2/thread.hpp' file not found
#include <boost/thread/v2/thread.hpp>

**Comment by mike2lab** [comment]
Ok nice :)

**Comment by mike2lab** [comment]
I use OS X 10.11.6

**Comment by valiant1x** [comment]
The issue is most likely cmake version, as each version of cmake only knows to look for the most up-to-date version of boost at the time cmake was released. Which version of cmake are you using? Did you try updating to latest?

**Comment by mike2lab** [comment]
I have cmake version 3.11.1

**Comment by iedemam** [comment]
Fixed the header path issue.

There is still a compilation issue in 1.67

````
[ 58%] Building CXX object src/wallet/CMakeFiles/obj_wallet.dir/api/wallet.cpp.o
/Users/michael/Documents/ITNS/GitHub/intensecoin-valiant1x/src/wallet/api/wallet.cpp:1296:45: error: no matching constructor for initialization of
      'boost::posix_time::milliseconds' (aka 'subsecond_duration<boost::posix_time::time_duration, 1000>')
            boost::posix_time::milliseconds wait_for_ms(m_refreshIntervalMillis);
                                            ^           ~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/date_time/time_duration.hpp:270:30: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from
      'std::atomic<int>' to 'const boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000>' for 1st argument
  class BOOST_SYMBOL_VISIBLE subsecond_duration : public base_duration
                             ^
/usr/local/include/boost/date_time/time_duration.hpp:270:30: note: candidate constructor (the implicit move constructor) not viable: no known conversion from
      'std::atomic<int>' to 'boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000>' for 1st argument
/usr/local/include/boost/date_time/time_duration.hpp:286:59: note: candidate template ignored: disabled by 'enable_if' [with T = std::__1::atomic<int>]
                                typename boost::enable_if<boost::is_integral<T>, void>::type* = 0) :
````

...which looks like it's related to this fix in Boost which limits to integral types: https://github.com/boostorg/date_time/pull/64.

Monero has had it reported as well here: https://github.com/monero-project/monero-gui/issues/1335

**Comment by iedemam** [comment]
I think I have it fixed, waiting on test builds now. Pull the boost-1.67 branch and give it a try if you have a chance too.

**Comment by iedemam** [comment]
Merged into master.
